tests/utils.py

```diff
@@ -18,3 +22,90 @@ def override_log_level(log_level: str):
     finally:
         # Reset the logger so we don't have verbose output in all unit tests
         logging.getLogger("httpx").handlers = []
+
+
+class MockTransport(httpcore.SyncHTTPTransport):
+    def __init__(self, handler: Callable) -> None:
+        self.handler = handler
+
+    def request(
+        self,
+        method: bytes,
+        url: Tuple[bytes, bytes, Optional[int], bytes],
+        headers: List[Tuple[bytes, bytes]] = None,
+        stream: httpcore.SyncByteStream = None,
+        timeout: Mapping[str, Optional[float]] = None,
+    ) -> Tuple[bytes, int, bytes, List[Tuple[bytes, bytes]], httpcore.SyncByteStream]:
+        raw_scheme, raw_host, port, raw_path = url
+        scheme = raw_scheme.decode("ascii")
+        host = raw_host.decode("ascii")
+        port_str = "" if port is None else f":{port}"
+        path = raw_path.decode("ascii")
+
+        request_headers = httpx.Headers(headers)
+        data = (
+            (item for item in stream)
+            if stream
+            and (
+                "Content-Length" in request_headers
+                or "Transfer-Encoding" in request_headers
+            )
```

Comment:
It's a bit fiddly to explain why we're having to do this at the moment, but it's related to some of the grungy bits that need ironing out in #1253.

I'll resolve it as part of a follow-up.